### PR TITLE
Add sample terraform config

### PR DESCRIPTION
**Pull request by nellshamrell**
#28 should be merged before this one is.

This adds a sample terraform configuration file for configuring this both in AWS ECS. This is meant to be a sample/reference terraform config (creating it was very helpful for me understand how to set up a bot in ECS), which I believe is clear from it being in the "terraform_example" directory and the README within that example directory.

The actual config we use for the production instance of this bot will live in [rust-lang/simpleinfra].

Why include this sample config if the actual config lives elsewhere? The simpleinfra directory is much harder to navigate for someone inexperienced directly with the Rust team infrastructure than a sample terraform config where everything is in one file. It makes more sense to have things modularized when dealing with multiple terraform configs at once like we do in simpleinfra - it's definitely the right thing to do in that situation. However, when it comes to learning how to use Terraform resources, it's often more helpful to initially have things in one config file.

I want to include this in case anyone comes across this bot and is interested in forking and customizing it for their purposes. This may seem unlikely, and perhaps it is, but I do think it's one of our duties as OSS maintainers to make the ability to fork and customize a project as easy as possible. Additionally, I often search GitHub for examples of code that does what I want to do - having this sample config committed in this GitHub repo would help others doing those kinds of searches. I do not believe we lose anything by having this available in our repo, and there is definitely potential gain both for us and for others who want to understand how this bot can work on an infrastructure level.

### CODE REVIEW

File: terraform_example/main.tf [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+  })  
+}
+
+resource "aws_iam_role_policy_attachment" "read_github_token" {
+  depends_on = [aws_iam_policy.read_github_token]
+  role = aws_iam_role.crate-io-ops-bot-task-execution-role.name
+  policy_arn = aws_iam_policy.read_github_token.arn
+}
+
+# Sets up ECS roles and the ECS cluster:w
```
Comment (pietroalbini) [inline terraform_example/main.tf:377]: Vim :D

**Comment by pietroalbini** [comment]
I'm still not convinced merging this in the repository is the right move.

> I want to include this in case anyone comes across this bot and is interested in forking and customizing it for their purposes. This may seem unlikely, and perhaps it is, but I do think it's one of our duties as OSS maintainers to make the ability to fork and customize a project as easy as possible.

I agree with you that making projects easy to fork and customize is a priority, but I don't think documenting and provide examples for *three* ways of deploying the code is needed: we already documented how to set it up with Docker and Heroku. ECS is just a layer on top of Docker, and the only thing unique to our bot in it is the list of environment variables in the task definition.

Also, it doesn't make much sense to deploy the bot on ECS if you don't already have other things in it as well. I ran some number, and this setup is 110$/month excluding bandwidth, most of which is the NAT gateways and the load balancer. If a person forks the bot and doesn't use ECS they should put it on Heroku, while if they already use ECS they already have the configuration up and they just need to customize the variables in the task definition.

> Additionally, I often search GitHub for examples of code that does what I want to do - having this sample config committed in this GitHub repo would help others doing those kinds of searches.

I think to achieve that goal it would be better to create a "deploy a bot on ECS" separate repository in a personal account, possibly with some documentation on why the choices were made :)

> I do not believe we lose anything by having this available in our repo, and there is definitely potential gain both for us and for others who want to understand how this bot can work on an infrastructure level.

There is some maintenance required for it, unless we want to just dump the code in the repository and forget about it. For example, if we add more environment variables in the future we'll need to update the task definition as well.

**Comment by smarnach** [comment]
I agree with Pietro – almost none of the Terraform code is specific to this bot, so it would be better to put it in its own repo.

**Comment by nellshamrell** [comment]
Understood and thank you for the thoughtful feedback. Closing this PR :)

**Pull request closed without merging**